### PR TITLE
[libspatialite] Update 5.0.0

### PR DESCRIPTION
**Pull request by longhuan2018**
update libspatialite to 5.0.0

### CODE REVIEW

**Comment by longhuan2018** [comment]
Both libspatialite and spatialite-tools depend on librttopo，After librttopo PR#15037 is merged, I will merge the code again.

**Comment by longhuan2018** [comment]
The modification of spatialite-tools is to make this PR static library compile without error, it will be updated in another PR #15039

**Comment by longhuan2018** [comment]
@JackBoosY Can anyone review this PR？

File: ports/libspatialite/fix-latin-literals.patch [hunk 1]
```diff
@@ -20,7 +20,7 @@ index a5c8334..dd75dde 100644
      add_srs_wkt(p,11,"PARAMETER[\"false_easting\",0],PARAMETER[\"false_northing\",0],");
      add_srs_wkt(p,12,"AUTHORITY[\"mj10777.de\",\"187916\"],AXIS[\"x\",NORTH],AXIS[\"y\",EAST]]");
 -    p = add_epsg_def(filter,first,last,187918,"mj10777.de",187918,"DHDN / Soldner 18 Müggelberg 600");
-+    p = add_epsg_def(filter,first,last,187918,u8"mj10777.de",187918,"DHDN / Soldner 18 M\u00fcggelberg 600");
++    p = add_epsg_def(filter,first,last,187918,"mj10777.de",187918,u8"DHDN / Soldner 18 M\u00fcggelberg 600");
```
Comment (JackBoosY) [inline ports/libspatialite/fix-latin-literals.patch:23]: Should this report to the upstream?

File: ports/libspatialite/fix-makefiles.patch [hunk 2]
```diff
@@ -54,8 +37,8 @@ index 0960ec1..276bc84 100644
 +INSTDIR=$(INST_DIR)
  
  # Uncomment the first for an optimized build, or the second for debug.
--OPTFLAGS=	/nologo /Ox /fp:precise /W3 /MD /D_CRT_SECURE_NO_WARNINGS \
-+OPTFLAGS=	/nologo /fp:precise /W3 $(CL_FLAGS) /D_CRT_SECURE_NO_WARNINGS \
+-OPTFLAGS=	/nologo /Ox /fp:precise /W4 /MD /D_CRT_SECURE_NO_WARNINGS \
++OPTFLAGS=	/nologo /fp:precise /W4 $(CL_FLAGS) /D_CRT_SECURE_NO_WARNINGS \
```
Comment (JackBoosY) [inline ports/libspatialite/fix-makefiles.patch:41]: I think we should use `W3` instead.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libspatialite/fix-latin-literals.patch (hunk 1 above)
Comment (longhuan2018) [reply ports/libspatialite/fix-latin-literals.patch]: I kept the file fix-latin-literals.patch from the previous version. I tried to delete it and found that the compilation was normal. Maybe it is not a necessary option now. So I deleted it.

File: ports/libspatialite/fix-makefiles.patch (hunk 2 above)
Comment (longhuan2018) [reply ports/libspatialite/fix-makefiles.patch:41]: W3 is changed to W4 in upstream, so I keep it consistent

**Review by JackBoosY — APPROVED** [review]

**Review by JackBoosY — CHANGES_REQUESTED** [review]
This port installed the pkgconfig file on non-Windows, so I think we should add `vcpkg_fixup_pkgconfig()` after `vcpkg_install_make` when building non-Windows.

**Review by JackBoosY — APPROVED** [review]

**Review by vicroms — APPROVED** [review]

**Comment by vicroms** [comment]
Thanks for the PR!

**Comment by JackBoosY** [comment]
@longhuan2018  Hi, yesterday I got a regression about `spatialite-tools`:
```
/home/usr/work/vcpkg/installed/x64-linux/debug/lib/libspatialite.a(lwn_network.o): In function `geo_link_split':
lwn_network.c:(.text+0x26e7): undefined reference to `ptarray_construct'
lwn_network.c:(.text+0x2791): undefined reference to `ptarray_set_point4d'
lwn_network.c:(.text+0x27d2): undefined reference to `rtline_construct'
lwn_network.c:(.text+0x27fc): undefined reference to `ptarray_construct'
lwn_network.c:(.text+0x2860): undefined reference to `ptarray_set_point4d'
lwn_network.c:(.text+0x2884): undefined reference to `rtpoint_construct'
...
```
This seems to be caused by libspatialite using **postgis** when building, but not adding the postgis library to the link list when building spatialite-tools.
But I did not find the postgis library in the build logs of libspatialite.
Do you have any ideas?

**Comment by longhuan2018** [comment]
> @longhuan2018 Hi, yesterday I got a regression about :`spatialite-tools`
> 
> ```
> /home/usr/work/vcpkg/installed/x64-linux/debug/lib/libspatialite.a(lwn_network.o): In function `geo_link_split':
> lwn_network.c:(.text+0x26e7): undefined reference to `ptarray_construct'
> lwn_network.c:(.text+0x2791): undefined reference to `ptarray_set_point4d'
> lwn_network.c:(.text+0x27d2): undefined reference to `rtline_construct'
> lwn_network.c:(.text+0x27fc): undefined reference to `ptarray_construct'
> lwn_network.c:(.text+0x2860): undefined reference to `ptarray_set_point4d'
> lwn_network.c:(.text+0x2884): undefined reference to `rtpoint_construct'
> ...
> ```
> 
> This seems to be caused by libspatialite using **postgis** when building, but not adding the postgis library to the link list when building spatialite-tools.
> But I did not find the postgis library in the build logs of libspatialite.
> Do you have any ideas?
> 
> Note the symbols are belong to postgis instead of geos. See [documentation](https://www.postgis.net/docs/doxygen/2.4/da/de7/liblwgeom_8h_a6967ac4744c94dc2cbbf9e43d0ae9d66.html).

Now libspatialite depends on librttopo, so if libspatialite is statically linked, librttopo also needs to be linked

**Comment by longhuan2018** [comment]
How can I reproduce this compilation error? Do need to download the latest vcpkg?

**Comment by JackBoosY** [comment]
@longhuan2018  Yes, do you have linux machine? Just install `spatialite-tools:x64-linux` can repro this issue.

**Comment by longhuan2018** [comment]
> @longhuan2018 Yes, do you have linux machine? Just install `spatialite-tools:x64-linux` can repro this issue.
OK, I'll try first

**Comment by JackBoosY** [comment]
@longhuan2018 Note #15746 trying to disable `spatialite-tools` results in pipeline test, so if your PR can successfully fix this regression, please leave a message in #15746 to close that PR.

**Comment by longhuan2018** [comment]
ok

**Pull request merged**